src/components/audio-params.js

```diff
@@ -0,0 +1,271 @@
+        this.el.parentEl?.parentEl?.querySelector("[avatar-volume-controls]").components["avatar-volume-controls"]?.data
+          .volume || 1.0;
+      this.el.parentEl?.parentEl?.addEventListener("avatar-volume-changed", this.onVolumeUpdated);
+    } else if (this.el.components["audio-target"]) {
+      this.data.sourceType = SourceType.AUDIO_TARGET;
+    }
+  },
+
+  remove() {
+    this.normalizer = null;
+    this.el.sceneEl?.systems["audio-debug"].unregisterSource(this);
+    if (!this.data.isLocal) {
+      this.el.sceneEl?.systems["hubs-systems"].gainSystem.unregisterSource(this);
+    }
+
+    this.el.removeEventListener("media-volume-changed", this.onVolumeUpdated);
+    this.el.parentEl?.parentEl?.removeEventListener("avatar-volume-changed", this.onVolumeUpdated);
+
+    if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+      this.el.components["avatar-audio-source"].el.removeEventListener("sound-source-set", this.onSourceSetAdded);
+    }
+  },
+
+  tick() {
+    const audio = this.audio();
+    if (audio) {
+      if (audio.updateMatrixWorld) {
+        audio.updateMatrixWorld(true);
+      }
+      this.data.position = new THREE.Vector3(
+        audio.panner.positionX.value,
+        audio.panner.positionY.value,
+        audio.panner.positionZ.value
+      );
+      this.data.orientation = new THREE.Vector3(
+        audio.panner.orientationX.value,
+        audio.panner.orientationY.value,
+        audio.panner.orientationZ.value
+      );
+      this.data.distanceModel = audio.panner.distanceModel;
+      this.data.rolloffFactor = audio.panner.rolloffFactor;
+      this.data.refDistance = audio.panner.refDistance;
+      this.data.maxDistance = audio.panner.maxDistance;
+      this.data.coneInnerAngle = audio.panner.coneInnerAngle;
+      this.data.coneOuterAngle = audio.panner.coneOuterAngle;
+      this.data.coneOuterGain = audio.panner.coneOuterGain;
+      this.updateDistances();
+      this.updateAttenuation();
+    }
+
+    if (this.normalizer !== null) {
+      this.normalizer.apply();
+    } else {
+      // We one only enable the Normalizer for avatar-audio-source components
+      if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+        this.enableNormalizer();
+      }
+    }
+  },
+
+  audio() {
+    if (this.data.sourceType === SourceType.AVATAR_RIG) {
+      // Create fake parametes for the avatar rig as it doens't have an audio source.
+      const audioParams = this.el.sceneEl.systems["hubs-systems"].audioSettingsSystem.audioSettings;
+      const avatarRigObj = document.getElementById("avatar-rig").querySelector(".camera").object3D;
+      avatarRigObj.updateMatrixWorld(true);
+      avatarRigObj.getWorldPosition(this.avatarRigPosition);
+      this.avatarRigOrientation.set(0, 0, -1);
+      const worldQuat = new THREE.Quaternion();
+      avatarRigObj.getWorldQuaternion(worldQuat);
+      this.avatarRigOrientation.applyQuaternion(worldQuat);
+      return {
```

Comment:
This is allocating several new objects per avatar every frame. I think that is going to lead to noticeable GC pressure.

Reply:
Moved this to a temp var to avoid allocation